site/docs/api/error-handling.md

```diff
@@ -0,0 +1,136 @@
+interesting part.
+
+## Types of Errors
+
+Before we dig deeper, let's list some typical error one might encounter:
+
+* *corrupt data*: ex. user-provided code is parsing JSON lines, but some
+  of them are incorrectly formatted
+* *bugs in user code*: ex. user provided filtering lambda throws a
+  `NullPointerException` for certain input
+* *connection to external services lost*: Jet sink writing into an
+  `IMap` loses connection to the remote Hazelcast cluster hosting the
+  data
+
+## Specific Solutions
+
+Some concrete types of error have solutions specifically tailored to
+address them.
+
+### In-Memory Data Structures
+
+Jet comes out of the box with some [in-memory distributed data
+structures](data-structures) which can be used as a data source or a
+sink. This are hosted by Hazelcast clusters, which can be the same as
+the one Jet runs on or separate, remote ones.
+
+In this case, all error handling (for example, communications failures
+with the remote cluster, but not just) is handled transparently to Jet,
+by the Hazelcast cluster itself. For further details and configuration
+options, read the [relevant section in the Hazelcast
+manual](https://docs.hazelcast.org/docs/latest/manual/html-single/#handling-failures).
+
+### Individual Sources & Sinks
+
+Sources and Sinks are Jet's points of contact with external data stores.
+Their various types have specific characteristics that enable error
+handling strategies applicable only to them.
+
+For example, our [Change Data Capture
+sources](sources-sinks#change-data-capture-cdc) can attempt to reconnect
+automatically whenever they lose connection to the databases they
+monitor, so for intermittent network failures, their owner jobs don't
+need to fail.
+
+For further such specific failure handling options, consult the Javadoc
+of Jet's various [sources and sinks](sources-sinks).
+
+## Generic Solutions
+
+In Jet we also have generic options for dealing with random errors
+affecting our jobs.
+
+### Jobs Without Mutable State
+
+For many streaming jobs, specifically the ones which don't have any
+processing guarantee configured, the pipeline definition and the job
+config are the only parts we can identify as state, and those are
+immutable.
+
+Batch jobs also fall into the category of immutable state jobs.
+
+One option for dealing with failure in immutable state jobs is simply
+restarting them (once the cause of the failure has been addressed).
+Restarted streaming jobs lacking mutable state can just resume
+processing the input data flow from the current point in time, batch
+jobs can be re-run from the beginning.
+
+### Processing Guarantees
+
+Jobs with mutable state, the ones that have [processing
+guarantees](../architecture/fault-tolerance#processing-guarantee-is-a-shared-concern)
+set, achieve their fault tolerance by periodically saving [recovery
+snapshots](../architecture/fault-tolerance#distributed-snapshot). When
```

Comment:
So why do the jobs snapshot? It feels that there is no benefit unless one configures suspend on failure

It's important to say that snapshots are useful to overcome member failures.

Reply:
I will emphasize that.